build(root): Update Jarvis to run docker infra instead of through brew

Cliftonz opened the pull request.

Cliftonz committed:
Refactor database setup to utilize Docker

The previous code that checked homebrew and installed databases if not found has been modified. Now, the script checks if MongoDB and Redis are installed, and if they aren't, it starts Docker Compose to run the databases. It also copies the example environment file to a specified location and starts up the Docker Compose. This improves the local dev environment setup by using Docker for database management.

Cliftonz committed:
Refactor dev environment setup script

The script for setting up the development environment has been updated for better readability and organization. Changes include removing redundant messages, moving Docker start-up into a designated function, and changing the Docker Compose command to start in detached mode. The script also includes success messages after starting Docker and verifying git installation.

rifont commented on scripts/dev-environment-setup.sh:
```diff
@@ -383,43 +389,29 @@ install_aws_cli () {
-    SKIP="$(check_homebrew)"
+start_database() {
+  # Check if mongodb is installed
+  brew ls --versions mongodb > /dev/null
```
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** What is the behaviour when `brew` is not installed on the developer's system? I think we should handle cases where it is not in a silent way.‏

rifont commented on scripts/dev-environment-setup.sh:
```diff
@@ -383,43 +389,29 @@ install_aws_cli () {
+  brew ls --versions mongodb > /dev/null
+  if [ $? -eq 0 ]; then
+    echo "Error: MongoDB is already installed via brew. Please uninstall it first."
+    exit 1
```
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
⚠ **issue:** I think throwing an error here could be a bit heavy-handed. I think there are valid cases for having Mongo/Redis installed in Brew for use outside of the Novu project. Perhaps we can log a warning here instead?‏

Cliftonz committed:
Merge branch 'next' into update-javis-to-use-docker

Cliftonz committed:
Update dev environment setup script for MongoDB and Redis checks

The development environment setup script has been updated to handle scenarios where MongoDB and Redis have already been installed. Rather than exiting, it now sets a flag and proceeds if both services aren't installed already, copying the environment file and starting Docker Compose accordingly.

Cliftonz committed:
Add brew installation check before starting database

The updated developer environment setup script now checks for availability of the brew command before trying to start the database. This provides cleaner handling for environments where brew is not available. If brew command is not found, the script checks the default ports for MongoDB and Redis instead.

Cliftonz committed:
Merge remote-tracking branch 'origin/update-javis-to-use-docker' into update-javis-to-use-docker

Cliftonz committed:
Add brew installation check before starting database

The updated developer environment setup script now checks for availability of the brew command before trying to start the database. This provides cleaner handling for environments where brew is not available. If brew command is not found, the script checks the default ports for MongoDB and Redis instead.

Cliftonz committed:
Merge branch 'next' into update-javis-to-use-docker

Cliftonz committed:
Merge branch 'next' into update-javis-to-use-docker

Cliftonz committed:
Merge branch 'next' into update-javis-to-use-docker

Cliftonz committed:
Merge branch 'next' into update-javis-to-use-docker

Cliftonz committed:
Merge branch 'next' into update-javis-to-use-docker

rifont reviewed (commented):
When running `pnpm dev-environment-setup`, I am getting a prompt to enter my password after the Redis installation check occurs, but I'm not made aware of why I need to enter my password. Is there a step in the setup sequence that we either need to:
1. Remove to prevent asking for password
2. Add an addition `echo` to give info to the User on _why_ they need to enter their password

```bash
# ...
✅ NVM has been installed
 
 
✅ Node.js v20.8.1 is already installed
 
 
✅ PNPM 8.9.0 is already installed
 
 
✅ Docker is already installed
 
 
✅ AWS CLI is already installed
 
Warning: Redis is already installed via brew. Please uninstall it first.
Password:
```

Cliftonz committed:
Update dev-environment-setup.sh to add messaging for manual database start

Updated the dev-environment-setup.sh script to include messages for manual database start. This includes a recommendation for removing mongodb and redis databases from brew and instructions on how to manually start the containerized databases in the project.

Cliftonz committed:
Remove local dev domain setup from dev environment script

This commit removes the function create_local_dev_domain from the development environment setup script. It was no longer necessary as it was causing conflicts with other processes. This would ensure the script runs smoother and cause less conflicts while setting up the development environment.

Cliftonz committed:
Reformatted code and updated Homebrew upgrade command

The code in the dev-environment-setup.sh script has been reformatted for more readability and consistency. Additionally, the command for upgrading Homebrew has been simplified. The previous version was tapping into Homebrew cask unnecessarily before updating, this step has now been removed.

Cliftonz committed:
Merge branch 'next' into update-javis-to-use-docker

rifont reviewed (commented):
⚡🚀👨‍💻

rifont reviewed (approved):
⚡🚀👨‍💻
